### DISCUSSION TITLE
What would be the best way to get a table of tailwindcss class to the rules it represents?

### DISCUSSION THREAD

**Discussion by vedantroy** [tailwindlabs/tailwindcss#11490] [General]
Hi! I was wondering what would be the best way to get a table of tailwindcss class to the rules it represents.
Is this data stored somewhere in the tailwindcss compiler?

**Comment by Robertopera** [comment]
The Tailwind CSS framework does not provide an official way to retrieve a table of all the class names and their corresponding CSS rules out of the box. The class-to-rule mapping is an internal implementation detail of Tailwind CSS and is not exposed as a direct API.

However, you can generate a JSON representation of the class-to-rule mapping by utilizing the @tailwindcss/visually-hidden plugin, which is included with Tailwind CSS. This plugin generates utility classes that visually hide elements on the screen while keeping them accessible to assistive technologies.

To generate the class-to-rule mapping, you can follow these steps:

Install the @tailwindcss/visually-hidden plugin as a development dependency in your project:

npm install --save-dev @tailwindcss/visually-hidden

Update your Tailwind CSS configuration (tailwind.config.js) to include the @tailwindcss/visually-hidden plugin:

module.exports = {
  // ...
  plugins: [
    require('@tailwindcss/visually-hidden'),
    // Other plugins...
  ],
};

Run your project's build command to generate the CSS file that includes the visually hidden utility classes.

Once the build process is complete, you can inspect the generated CSS file to find the visually hidden utility classes. These classes will have unique CSS rules associated with them, representing the class-to-rule mapping.

It's worth noting that the class-to-rule mapping may not be directly retrievable in a structured format like JSON. You may need to manually parse the generated CSS file or use a CSS parsing library to extract the necessary data.

Keep in mind that this approach relies on the @tailwindcss/visually-hidden plugin, and the generated CSS may include additional utility classes beyond the visually hidden ones. Additionally, this method will only provide you with the class-to-rule mapping for the utility classes included in Tailwind CSS, not for any custom classes or styles you may have defined.
 [YourTexasBenefits Application](https://www.yourtexasbenefits.bid/)
